handlers/cache.js

```diff
@@ -30,15 +29,54 @@ var cacheTypeIDs = {
     CSODOTALobbyInvite : 2011
 };
 
+
+
 // Handlers
+function handleCreateType(obj_type, object_data) {
+    switch(obj_type) {
+        case cacheTypeIDs.CSOEconItem:
+            this.Logger.debug("Got an item traded");
+            var item = Dota2.schema.lookupType("CSOEconItem").decode(object_data);
+            this.emit("gotItem", item);
+            this.Inventory.push(item);
+            break;
+    }
+}
+
+function handleDestroyType(obj_type, object_data) {
+    switch(obj_type) {
+        case cacheTypeIDs.CSODOTAPartyInvite:
+            this.Logger.debug("Party invite cleared");
+            this.PartyInvite = null;
+            this.emit("partyInviteCleared");
+            break;
+        case cacheTypeIDs.CSODOTALobbyInvite:
+            this.Logger.debug("Lobby invite cleared");
+            this.LobbyInvite = null;
+            this.emit("lobbyInviteCleared");
+            break;
+        case cacheTypeIDs.CSOEconItem:
+            this.Logger.debug("Traded item away");
+            var item = Dota2.schema.lookupType("CSOEconItem").decode(object_data);
+            this.Inventory = this.Inventory.filter(i => item.id.equals(i.id));
```

Comment:
Use `notEquals` (or `! equals`) if you want to remove a _single_ item from an array.

In addition:
> Note that the `Inventory` property will be the old value until after this event completes to allow comparison between the two.

So you probably have to call `emit` _before_ `filter`.